### PR TITLE
AdminUI - Add SearchKit-based screen for Administer Location Types

### PR DESCRIPTION
**Pull request by wmortada**
Overview
----------------------------------------
This converts the 'Administer Location Types' screen to use SearchKit.

Before
----------------------------------------
image

After
----------------------------------------
image

Comments
----------------------------------------
Alters the behaviour of CiviCRM core so that the edit form is on a separate URL. This is necessary to enable us to replace the listing screen but allow the update and delete actions to work as previously.

I've made a couple of improvements to the original screen. I've simplified the title to just 'Location Types' (rather than 'Location Types (Home, Work...)' and I've added a tooltip to the first column to show the ID for those that need this.

There is a question about whether we need both a 'Name' and a 'Display Name' for Location Types but this is out of scope of this PR.

### CODE REVIEW

**Comment by wmortada** [comment]
TODO: Hide the delete button for 'Billing'

File: ext/civicrm_admin_ui/managed/SavedSearch_Administer_Location_Types.mgd.php [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            ],
+            [
+              'type' => 'field',
+              'key' => 'is_default',
+              'dataType' => 'Boolean',
+              'label' => E::ts('Default?'),
+              'sortable' => TRUE,
+            ],
+            [
+              'size' => 'btn-sm',
```
Comment (colemanw) [inline ext/civicrm_admin_ui/managed/SavedSearch_Administer_Location_Types.mgd.php:119]: Should be
```suggestion
              'size' => 'btn-xs',
```

File: ext/civicrm_admin_ui/managed/SavedSearch_Administer_Location_Types.mgd.php [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+              'dataType' => 'String',
+              'label' => E::ts('Description'),
+              'sortable' => TRUE,
+              'editable' => TRUE,
+            ],
+            [
+              'type' => 'field',
+              'key' => 'is_active',
+              'dataType' => 'Boolean',
+              'label' => E::ts('Enabled?'),
```
Comment (colemanw) [inline ext/civicrm_admin_ui/managed/SavedSearch_Administer_Location_Types.mgd.php:107]: In the other PRs we've been removing these rather silly question marks, so for consistency let's do it here too.
```suggestion
              'label' => E::ts('Enabled'),
```

File: ext/civicrm_admin_ui/managed/SavedSearch_Administer_Location_Types.mgd.php [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+              'sortable' => TRUE,
+              'editable' => TRUE,
+            ],
+            [
+              'type' => 'field',
+              'key' => 'is_default',
+              'dataType' => 'Boolean',
+              'label' => E::ts('Default?'),
```
Comment (colemanw) [inline ext/civicrm_admin_ui/managed/SavedSearch_Administer_Location_Types.mgd.php:115]:
```suggestion
              'label' => E::ts('Default'),
```

File: ext/civicrm_admin_ui/managed/SavedSearch_Administer_Location_Types.mgd.php [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+            'text' => E::ts('Add Location Type'),
+            'icon' => 'fa-plus',
+          ],
+          'cssRules' => [
+            [
+              'disabled',
+              'is_active',
+              '=',
+              FALSE,
+            ],
```
Comment (colemanw) [inline ext/civicrm_admin_ui/managed/SavedSearch_Administer_Location_Types.mgd.php:163]: Great! CSS rule is a nice touch. We should make sure the others do this as well.

File: ext/civicrm_admin_ui/managed/SavedSearch_Administer_Location_Types.mgd.php [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+          ],
+          'columns' => [
+            [
+              'type' => 'field',
+              'key' => 'name',
+              'dataType' => 'String',
+              'label' => E::ts('Name'),
+              'sortable' => TRUE,
+              'title' => 'Location Type ID: [id]',
```
Comment (colemanw) [inline ext/civicrm_admin_ui/managed/SavedSearch_Administer_Location_Types.mgd.php:77]:
```suggestion
              'title' => E::ts('Location Type ID: [id]'),
```

File: ext/civicrm_admin_ui/ang/afsearchAdministerLocationTypes.aff.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
+    "type": "search",
+    "title": "Location Types",
+    "icon": "fa-list-alt",
+    "server_route": "civicrm/admin/locationType",
+    "permission": "access CiviCRM"
```
Comment (colemanw) [inline ext/civicrm_admin_ui/ang/afsearchAdministerLocationTypes.aff.json:6]:
```suggestion
    "permission": "administer CiviCRM"
```
Also, this file needs a newline at the end of the file.

File: ext/civicrm_admin_ui/managed/SavedSearch_Administer_Location_Types.mgd.php (hunk 4 above)
Comment (wmortada) [reply ext/civicrm_admin_ui/managed/SavedSearch_Administer_Location_Types.mgd.php:163]: The buttons don't look great when disabled, but I think that's a separate issue.

image

**Comment by colemanw** [comment]
I've reviewed this and it looks complete. Code looks perfect. Good work @wmortada

**Comment by wmortada** [comment]
Thanks @colemanw

File: ext/civicrm_admin_ui/managed/SavedSearch_Administer_Location_Types.mgd.php (hunk 4 above)
Comment (wmortada) [reply ext/civicrm_admin_ui/managed/SavedSearch_Administer_Location_Types.mgd.php:163]: I've created an issue for this: https://lab.civicrm.org/dev/core/-/issues/3916

**Pull request merged**